templates/adp-backend-template-dotnet/skeleton/src/${{ values.dotnet_solution_name }}.Api/Program.cs

```diff
@@ -0,0 +1,116 @@
+
+namespace ${{ values.dotnet_solution_name }}.Api
+{
+    [ExcludeFromCodeCoverage]
+    public static class Program
+    {
+        public static void Main(string[] args)
+        {
+            var builder = WebApplication.CreateBuilder(args);
+
+            // Add services to the container.
+            ConfigureServices(builder);
+
+            var app = builder.Build();
+
+            // Configure the HTTP request pipeline.
+            if (app.Environment.IsDevelopment())
+            {
+                app.UseSwagger();
+                app.UseSwaggerUI();
+                app.UseDeveloperExceptionPage();
+            }
+            app.UseExceptionHandler();
+
+            app.UseHttpsRedirection();
+            app.UseAuthorization();
+            app.MapControllers();
+
+            app.UseHealthChecks("/healthy", new HealthCheckOptions()
+            {
+                Predicate = check => check.Name == "ready"
+            });
+            app.UseHealthChecks("/healthz", new HealthCheckOptions()
+            {
+                Predicate = check => check.Name == "live"
+            });
+            app.UseHealthChecks("/", new HealthCheckOptions()
+            {
+                Predicate = check => check.Name == "live"
+            });
+
+            app.Run();
+        }
+
+        public static void ConfigureServices(WebApplicationBuilder builder)
+        {
+            builder.Services.AddApplicationServices();
+
+            builder.Services.AddLogging();
+            builder.Services.AddExceptionHandler<GlobalExceptionHandler>();
+            builder.Services.AddProblemDetails();
+
+            builder.Services.AddHealthChecks()
+                   .AddCheck<LivenessCheck>("live")
+                   .AddCheck<ReadinessCheck>("ready");
+
+            var appInsightsConfig = new AppInsightsConfig
+            {
+                ConnectionString = builder.Configuration.GetValue<string>("AppInsights:ConnectionString") ?? string.Empty,
+                CloudRole = builder.Configuration.GetValue<string>("AppInsights:CloudRole") ?? string.Empty
+            };
+
+            ConfigureOpenTelemetry(builder, appInsightsConfig);
+
+            builder.Services.AddControllers();
+            builder.Services.AddEndpointsApiExplorer();
+            builder.Services.AddSwaggerGen(config =>
+            {
+                config.SwaggerDoc("v1", new OpenApiInfo { Title = "${{ values.dotnet_solution_name }}", Version = "v1" });
+            });
+            builder.Services.AddApiVersioning(config =>
+            {
+                config.DefaultApiVersion = new ApiVersion(1, 0);
+                config.AssumeDefaultVersionWhenUnspecified = true;
+                config.ReportApiVersions = true;
+                config.ApiVersionReader = new HeaderApiVersionReader("api-version");
+            });
+        }
+
+        private static void ConfigureOpenTelemetry(WebApplicationBuilder builder, AppInsightsConfig appInsightsConfig)
```

Comment:
Could this be moved to an extension method? Doesnt feel like it fits the program.cs